src/plugins/dashboard/public/dashboard_container/embeddable/api/run_save_functions.tsx

```diff
@@ -285,10 +285,12 @@ export async function runInteractiveSave(this: DashboardContainer, interactionMo
               lastSavedTitle: currentState.title,
               copyOnSave: true,
               isTitleDuplicateConfirmed: false,
+              onTitleDuplicate(speculativeSuggestion) {
+                newTitle = speculativeSuggestion;
+              },
             }))
           ) {
-            copyCount++;
-            newTitle = `${baseTitle} (${copyCount})`;
+            // see onTitleDuplicate
```

Comment:
Why do we still have this `while` loop? The function should only be run once. Can you remove the loop and the comment above?

Reply:
I'd kept the while loop, to still solve for a case where we might have for some reason were not able to suggest a collision free title then old behaviour kicks in, but we the revised approach I'm taking we have no need for it anymore